### DISCUSSION TITLE
Is there a way to detect if a link was dragged and dropped on the background ?

### DISCUSSION THREAD

**Discussion by slechanii** [xyflow/xyflow#1849] [Q&A]
Hi, i want to make it so when the user drags a link from a handle to connect it he can drop it on the background where a menu will appear.

I have found no way to detect if a user has dragged a node into the background (only on another handle) yet and would like to know if there is a way to detect this ?

Another solution would be if I could detect when a user has started dragging a link and when he released it (but it must be regardless of where it dropped it and trigger everytime) I didn't find anything that could do this yet as well.

Does somebody know if this feature exists or an alternative way to make it work ?

Thank you

**Comment by moklick** [comment]
You could set a flag when you start an edge update and check if that flag is still set when the update is finished: https://codesandbox.io/s/spring-cache-ctw7j?file=/src/Flow.js In this example I remove an edge when the edge update doesn't end with a new connection.

**Reply by slechanii** [reply]
Thanks for your answer, but none of these callbacks seem to trigger when I drag an edge in the background (not on a node nor on a handle)

I've tested your code with console.log and it seems it doesn't get triggered when dropping the edge on nothing.

Am I doing it wrong ?

**Reply by moklick** [reply]
I thought that this was your question.. Now that I am reading it again, it's about a normal connection from a handle. In that case I would create a new node type "node-creator". That node gets created and just displays a menu with your node types. When you select a type, the type of the node-creator gets changed to the selected type.

**Reply by slechanii** [reply]
That's a great solution but is there a way to make it work and trigger by the edge drop ?

Ideally I would take an edge and drop it in the background, and this "node creator" would appear where I dropped my edge but I cannot find a way to detect when I dropped an edge somewhere that is not a node or another handle.

My problem is about finding a way to detect when the user dragged an edge and dropped it on something that isn't connectable (the background)

**Reply by datoslabs** [reply]
@slechanii , when user takes an edge and drops it in the background, `onConnect` will not fire whereas if the user takes an edge and drops on a `node` handle, `onConnect` will fire.  Using the flag suggested by @moklick , would it allow you to build the desired UX described?
